### DISCUSSION TITLE
Feature Request: Default values for Plain Text and Rich Text fields

### DISCUSSION THREAD

**Discussion by robmcfadden** [craftcms/cms#8461] [Ideas] [answered]
When creating a new Plain Text or Rich Text field, it would be helpful to be able to optionally enter a default value. This could then be parsed for element variables as an object template.

## Use case: 
We have a client with over a thousand entries in a section and a couple fields that follow the same formula, with only the {title} being dynamic. The alternative is creating a Global or putting the default value right in the template.

@timkelty was talking about potentially making a plugin for this, but thought it might be a good candidate for a core addition.

**Comment by timkelty** [comment]
Ended up making this for a client (right now just Craft 2): https://github.com/timkelty/craft2-defaulttext

**Comment by benface** [comment]
@brandonkelly Would you accept a PR for this?

**Comment by brandonkelly** [comment]
@timkelty works here now so maybe we can just get him to PR it :)

**Comment by brandonkelly** [comment, marked as answer]
We just released Craft 3.7.29 which provides a solution for this: It’s now possible to bulk-set individual element attributes/custom field values via `resave/*` commands (see #10356). So for example, after creating a new Plain Text field, you could set a default value on all the entries in a `foo` section like so:

```sh
php craft resave/entries --section foo --set myPlainTextField --to "default value" --if-empty
```

**Reply by KarlBishop** [reply]
I'm not sure if this fully addresses the feature request...?

My reading of it at least is that it would be desirable to have a default value for when new Entries are _created_.

Using `resave/entries` only applies a default value to existing Entries, so CP users still have to manually enter the default value for each new Entry.

**Reply by KarlBishop** [reply]
Happy to add this as a separate Feature Request...?

**Reply by KarlBishop** [reply]
Also, I was wondering: when using resave/entries, how can I escape { and } in a raw value?

E.g. I want to do this:
````
php craft resave/entries --section foo --set myPlainTextField --to "=default value with {braces}" --if-empty
````

But the output is: "default value with "

**Reply by i-just** [reply]
If you want the output to be literally “default value with {braces}”, then you can try this:
```
php craft resave/entries --section foo --set myPlainTextField --to "fn(\$entry) => \"default value with {braces}\"" --if-empty
```

**Reply by KarlBishop** [reply]
> If you want the output to be literally “default value with {braces}”, then you can try this:
> 
> ```
> php craft resave/entries --section foo --set myPlainTextField --to "fn(\$entry) => \"default value with {braces}\"" --if-empty
> ```

Thanks, that worked a treat!
